File: 4-May Number Complement/Number_Complement_1.cpp
class Solution {
public:
    int findComplement(int num) {
        int count = 0;
        int i = num;
        while(i != 0){
            i = i/2;
            count++;
        }
        //printf("%d", count);
        int k = pow(2,count) - 1;
        //printf("%d",k);
        return num ^ k;
    }
};

File: 04-May Number Complement/Number_Complement_1.py
class Solution:
    def findComplement(self, num: int) -> int:
        dict1 = {'1':'0','0':'1'}
        x = bin(num)
        x=list(x[2:])
        for i in range(0, len(x)):
            x[i] = dict1[x[i]]
        x = "".join(x)
        s = int(x, 2)
        return s
